### PR TITLE
Add Deflate64 decompressor

### PR DESCRIPTION
**Pull request by cielavenir**
zipfile-deflate64 0.2.0 is now stable with my patch, so I think it can be included into py7zr.

it uses https://github.com/cielavenir/zlib_infback9_resumable/tree/infback9_resumable/contrib/infback9 , so the license should be safe.

/cc @brianhelba

### CODE REVIEW

File: py7zr/compressor.py [hunk 1]
```diff
@@ -31,6 +31,7 @@
 
 import pyppmd
 import pyzstd
+from zipfile_deflate64 import deflate64
```
Comment (brianhelba) [inline py7zr/compressor.py:34]: I wasn't intending for this to be a public / stable API, so this should have been named `_deflate64`.

However, I'd be happy to help with the usage of zipfile-deflate64 in this project, so I'm willing to make this actually a stake API. I may want to make some small revisions first (in another release). I should be able to do that in the next few days.

**Review by brianhelba — COMMENTED** [review]
This needs to add a package dependency on `zipfile-deflate64`. I'd suggest pinning to `0.2.*`, to avoid breakage from any future changes.

File: py7zr/compressor.py (hunk 1 above)
Comment (cielavenir) [reply py7zr/compressor.py:34]: @brianhelba omg, it is already used in https://github.com/cielavenir/zipfile39/blob/master/zipfile39.py#L77

Comment (cielavenir) [reply py7zr/compressor.py:34]: but ok, if minor revision I can follow it

**Comment by miurahr** [comment]
Please merge or rebase against master branch to fix known CI error #402

**Comment by cielavenir** [comment]
> Please merge or rebase against master branch to fix known CI error #402

done

**Comment by miurahr** [comment]
The error is caused by missing dependency.

>  py7zr/compressor.py:34: in <module>
      from zipfile_deflate64 import deflate64
  E   ModuleNotFoundError: No module named 'zipfile_deflate64'

Please add dependency to setup.cfg options section:

```ini

[options]
python_requires = >= 3.6
install_requires =
      texttable
      ....

```

You can specify version requirement.

**Comment by miurahr** [comment]
Please add test cases, (1) extract archive with DEFLATE64 (2) compress and decompress files with DEFLATE64, and check also whether p7zip can extract it.

**Comment by miurahr** [comment]
I've updated proposed branch. Please pull it from your repository.

1. Update dependency I've commented
2. Add test cases and update test expectations
3. Fix constant export

Because a test to compress and decompress data with DEFLATE64 failed, please fix the issue.

**Comment by cielavenir** [comment]
> a test to compress and decompress data with DEFLATE64 failed

zipfile_deflate64 only supports decompression.

in zipfile39 https://github.com/cielavenir/zipfile39/blob/master/test/test_zipfile.py, most of the tests check compress/decompress but deflate64 check decompressibility archive created by 7z.

It could be possible to add test using 7z, but without installing 7z testing would be difficult.

**Comment by cielavenir** [comment]
also @brianhelba what about the "stable" API?

**Comment by miurahr** [comment]
> > a test to compress and decompress data with DEFLATE64 failed
> 
> zipfile_deflate64 only supports decompression.

Thank you for feedback. I've updated the expectation to be raised UnsupportedCompressionMethod.
Please check a commit.

 
> in zipfile39 https://github.com/cielavenir/zipfile39/blob/master/test/test_zipfile.py, most of the tests check compress/decompress but deflate64 check decompressibility archive created by 7z.
> 
> It could be possible to add test using 7z, but without installing 7z testing would be difficult.

py7zr repository has already deflate64.7z test data in `tests/data/` folder and work well.

**Comment by miurahr** [comment]
@cielavenir  Could you fix a behavior when specify a deflate64 as a filter? The test failed because it does not raise an expected exception, but raises another exception.

File: py7zr/compressor.py [hunk 2]
```diff
@@ -1005,6 +1028,14 @@ class SupportedMethods:
             "filter_id": FILTER_BROTLI,
             "type": MethodsType.compressor,
         },
+        {
+            "id": COMPRESSION_METHOD.MISC_DEFLATE64,
+            "name": "DEFLATE64",
+            "native": False,
+            "need_prop": False,
+            "filter_id": FILTER_DEFLATE64,
```
Comment (miurahr) [inline py7zr/compressor.py:1036]: When compression is not supported,   ` "filter_id": FILTER_DEFLATE64,` should be `"filter_id": None`

**Comment by cielavenir** [comment]
@miurahr actually I'm thinking of applying https://github.com/cielavenir/python-codecs7z to provide deflate64 compression. It has been [already integrated into zipfile39](https://github.com/cielavenir/zipfile39).

**Comment by cielavenir** [comment]
@miurahr actually I'm thinking of applying https://github.com/cielavenir/python-codecs7z to provide deflate64 compression. It has been [already integrated into zipfile39](https://github.com/cielavenir/zipfile39).

/cc @brianhelba

**Comment by miurahr** [comment]
mypy raises error because zipfile_deflate64 don't have type definitions.

**Comment by cielavenir** [comment]
> zipfile_deflate64 don't have type definitions.

sorry where do you need it? I checked https://github.com/miurahr/py7zr/blob/master/py7zr/compressor.py but DeflateCompressor does not have type definitions.
or do you mean zipfile_deflate64/_zipfile.py raises error while zipfile_deflate64/deflate64 does not? if yes is it possible to exclude zipfile_deflate64/_zipfile.py ?

**Comment by miurahr** [comment]
Now passed all the test. I'd like to merge!

**Pull request merged**